book/onchain-verification/solidity-sdk.md

```diff
@@ -64,7 +64,14 @@ contract Fibonacci {
 
 ```
 
-For more details on the contracts, refer to the [sp1-contracts](https://github.com/succinctlabs/sp1-contracts) repo.
+### Finding your program vkey
+
+The program vkey (`fibonacciProgramVKey` in the example above) is passed into the `ISP1Verifer` along with the public values and proof bytes. You
+can find your program vkey by going through the following steps:
+
+1. Find what version of SP1 crates you are using.
+2. Use the version from step to run this command: `sp1up --version <version>`
+3. Use the vkey command to get the program vkey: `cargo prove vkey -elf <path/to/elf>`
```

Comment:
That's why step 1 exists, so that they ensure it matches

Reply:
Got it, still think that we should provide the two different ways that users can get the vkey.

Using `sp1up --version <version>` should return an error message when the version requested does not have published artifacts, as it currently doesn't work for `3.0.0-rc1` and `3.0.0-rc2`, because we never published artifacts for those.

Related: https://linear.app/succinct/issue/GRO-222/sp1up-verison-version-error-message